src/test/java/com/example/weather/client/wrapper/HttpRequestWrapperImplTest.java

```diff
@@ -0,0 +1,30 @@
+package com.example.weather.client.wrapper;
+
+import lombok.SneakyThrows;
+import org.junit.jupiter.api.Test;
+import org.springframework.http.HttpRequest;
+import org.springframework.http.client.support.HttpRequestWrapper;
+
+import java.net.URI;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class HttpRequestWrapperImplTest {
+
+    private final HttpRequest httpRequest = mock(HttpRequest.class);
+    private final String keyValue = "value";
+    private HttpRequestWrapperImpl httpRequestWrapperImpl = new HttpRequestWrapperImpl(httpRequest, keyValue);
+
+    @SneakyThrows
+    @Test
+    void testGetURI() {
+        HttpRequestWrapper httpRequestWrapper = mock(HttpRequestWrapper.class);
+        var uri = new URI(keyValue);
+        when(httpRequestWrapper.getURI()).thenReturn(uri);
+        var implURI = httpRequestWrapperImpl.getURI();
+        assertThat(implURI.toString()).contains(keyValue);
+        assertThat(implURI.toString()).isEqualTo("null&appid=" + keyValue);
```

Comment:
some things there should be done in a different way to properly reflect the logic in the class